### ISSUE TITLE
Unexpected table alias affixed to param when casting with ::

### ISSUE THREAD

**Issue by ddxv** [sqlfluff/sqlfluff#7570]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Given a postgresql snippet like:

```
with mysel as (select * from mytable mt
where mt.colname = :colname :: text
)
select * from mysel ms;
```

### Expected Behaviour

The expectation would be to format like as is.

### Observed Behaviour

but instead SQLFluff formats like:

```
with mysel as (
    select * from mytable as mt
    where mt.colname = mt.:colname::text
)

select * from mysel;
```

Where the alias `mt` is appeneded to the `:colname` placeholder.  This issue seems very closely related to https://github.com/sqlfluff/sqlfluff/issues/7293 which was recently fixed.

### How to reproduce

The main key I think in reproducing is including the cast with double colons `::` which seems to have trouble with the spacing as well. Without the `::` the format works correctly.

```
[tool.sqlfluff.core]
templater = 'placeholder'
dialect = 'postgres'

[tool.sqlfluff.templater.placeholder]
param_style = 'colon'
```

### Dialect

Postgresql. via pre-commit

### Version

Python3.14 and SQLFluff 4.04

### Configuration

```
[tool.sqlfluff.core]
templater = 'placeholder'
dialect = 'postgres'

[tool.sqlfluff.templater.placeholder]
param_style = 'colon'
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by ddxv** [comment]
I also just tried:

```
with mysel as (
    select * from mytable as mt
    where mt.colname = CAST(:mycol AS text)
)

select * from mysel;
```

And realized that this style of more explicit cast without the `::` did not resolve the issue, so I guess the title assumption that it was directly related to `:` param and `::` casting was incorrect.

**Comment by peterbud** [comment]
Hi @ddxv  thanks for submitting this issue. 

I think the second example is a better one to isolate the issue. Here is a minimal example:

```sql
select * from mytable as mt
where mt.colname = CAST(:colname AS TEXT)
```

when linted the 'AS' is triggering CP01 and is transformed to 'as' which is OK, but along the way the parameter placeholder is prefixed with the table name:

```sql
select * from mytable as mt
where mt.colname = CAST(mt.:colname as TEXT)
```
